node/src/database/db_initializer.rs

```diff
@@ -1033,7 +1157,175 @@ mod tests {
         }
         let subject = DbInitializerReal::default();
 
-        let _ = subject.initialize(&home_dir, true, MigratorConfig::panic_on_migration());
+        let _ = subject.initialize(
+            &home_dir,
+            true,
+            DbInitializationConfig::panic_on_migration(),
+        );
+    }
+
+    fn connection_special_setup_changed_assertion<F, E>(init_conn: Option<&Connection>, modifier: F)
```

Comment:
The big `if` statement in the middle of this function really makes it two separate functions. They probably ought to be separated and given different names, and perhaps `ask_pragma` and `ending_assertion` could be made external functions that are called from both of them.